### PR TITLE
Direct Line Web Chat Sample

### PR DESCRIPTION
**Pull request by peterinnesmsft**
### Purpose
#3215: Adding a sample web chat client that integrates using either Direct Line or Direct Line Speech depending upon application configuration.

### Changes
Adding new sample client project, DirectLine.Web, to the Clients solution.

### Tests
No tests as this is sample code only. Manually verified using both configurations.

### Feature Plan
N/A

#### General
- [x] I have commented my code, particularly in hard-to-understand areas	
- [x] I have added or updated the appropriate tests	
- [x] I have updated related documentation

### CODE REVIEW

**Comment by ryanisgrig** [comment]
@peterinnesmsft how did you validate the Enhanced DL flow locally? Did you have to publish a new web app to test against?

**Comment by peterinnesmsft** [comment]
> @peterinnesmsft how did you validate the Enhanced DL flow locally? Did you have to publish a new web app to test against?

Thankfully I did not have to create a new web app. Within the Web App Bot resource, I configured the Direct Line Channel to enable Enhanced Auth - as part of this, for testing a local app, you need to add http://localhost as a Trusted Origin (N.B.: even though it might be locally hosted using HTTPS, you explicitly need to set using HTTP, otherwise the origin presented in the token to ABS does not successfully find the corresponding session which is required as part of the auth flow).

Beyond this, I also used ngrok against locally running versions of the Skill and VA to verify that the activity containing the access token from the OAuth Prompt was indeed successfully sent back to the Skill, by updating the Web App Bot configs to point to the temporary ngrok endpoints. Otherwise, that was the extent of it.

**Review by lauren-mills — APPROVED** [review]

File: samples/csharp/clients/directline.web/DirectLine.Web.csproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.2.0" PrivateAssets="All" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Folder Include="wwwroot\images\" />
```
Comment (lzc850612) [inline samples/csharp/clients/directline.web/DirectLine.Web.csproj]: probably not needed

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: samples/csharp/clients/directline.web/DirectLine.Web.csproj (hunk 1 above)
Comment (peterinnesmsft) [reply samples/csharp/clients/directline.web/DirectLine.Web.csproj]: Removed in latest iteration.

**Review by darrenj — APPROVED** [review]
We should have an accompanying doc item guiding how to make use of this sample and background.

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**